pkg/cloudprovider/drift.go

```diff
@@ -73,3 +83,28 @@ func (c *CloudProvider) isAMIDrifted(ctx context.Context, machine *v1alpha5.Mach
 	}
 	return !lo.Contains(lo.Keys(mappedAMIs), instance.ImageID), nil
 }
+
+func (c *CloudProvider) isSecurityGroupDrifted(ec2Instance *instance.Instance, nodeTemplate *v1alpha1.AWSNodeTemplate) bool {
```

Comment:
Security groups can also be defined through an external launch template with launcTemplateName. I think this will break those users that are using that, so we need to make sure that we gate this drift to avoid drifting for those users